--- a/bolt/vendor/ircmaxell/random-lib/lib/RandomLib/Source/URandom.php	2014-02-13 13:35:25.781343500 +0000
+++ b/bolt/vendor/ircmaxell/random-lib/lib/RandomLib/Source/URandom.php	2014-01-03 11:26:13.000000000 +0000
@@ -54,7 +54,7 @@
      * @return string A string of the requested size
      */
     public function generate($size) {
-        if ($size == 0 || !file_exists($this->file)) {
+        if ($size == 0 || !@file_exists($this->file)) {
             return str_repeat(chr(0), $size);
         }
         $file = fopen($this->file, 'rb');
